### DISCUSSION TITLE
Root privileges not being passed down to child process when using npm scripts

### DISCUSSION THREAD

**Discussion by dhruvit-r** [npm/feedback#320] [CLI] [answered]
I came across `Error: listen EACCES: permission denied 0.0.0.0:80` when I tried to run an an http server on port 443. After some digging around, I figured out that it worked fine on npm 6. So, npm 7 was causing the problem. Also found  https://github.com/npm/cli/issues/2027 with a few different users facing the same issue. The issue was closed and I don't think the maintainers are monitoring it anymore, which is why I'm starting a new discussion here :)

1. I'm not sure about the reasons behind this change, but it probably has something to do with security, though I fail to see any reasons other than someone deciding that the user needs to be protected from himself or herself. If there has been a discussion around this which is available publicly, could you please point me to it? I'd really like to understand the thought process behind it. I wasn't able to find anything on the repo or the discussions. Just a place for me to start looking would suffice, thanks.
2. The documentation around this seems lacking. I found a single line in the change-log that would describes the cause of this issue. I'd be happy to contribute here in any way I can, but I think there needs to be a better explantation for the user. Maybe a message/log by npm when the user launches a script with admin privileges but the directory is owned by someone without admin privileges?
3. The workaround suggested in https://github.com/npm/cli/issues/2027 is to change the ownership of the directory to root. In my opinion, that is not a valid workaround. For me, it breaks a ton of stuff when I do this. Would it be possible to re-consider this decision? Or maybe allow the user to disable this behaviour using a flag in npm config?

**Comment by MylesBorins** [comment, marked as answer]
@dhruvit-r the @npm/cli are the ultimate decision makers about what happens in the CLI. If you would like to suggest an alternative to the suggestion of chowning the directory I suggest opening a proposal at https://github.com/npm/rfcs with an explicit outline of how you would like to see this problem fixed
